ouroboros-consensus/src/Ouroboros/Consensus/Block/SupportsMetrics.hs

```diff
@@ -0,0 +1,35 @@
+-- | See 'BlockSupportsMetrics'.
+module Ouroboros.Consensus.Block.SupportsMetrics (
+    BlockSupportsMetrics (..)
+  , WhetherSelfIssued (..)
+  , isSelfIssuedConstUnknown
+  ) where
+
+import           Ouroboros.Consensus.Block.Abstract
+
+{-------------------------------------------------------------------------------
+  Supported blocks
+-------------------------------------------------------------------------------}
+
+-- | A bespoke 3-valued boolean: yes, no, or unknown
+data WhetherSelfIssued =
```

Comment:
We use three different names for the same thing: issued, forged, produced. Unification on the name (picking one) would makes thing simpler and this amount of comments would not be needed at all, code would be crisp and clear.

Reply:
Yeah, this is pervasive, some very nebulous tech debt regarding consistent terminology (eg the research papers mean one thing by "fork" and "hard fork" uses the word completely differently).

For whatever reason, the ledger (eg the definition of header fields) seems to favor `issue` whereas the consensus code uses `forge` and moreover we've recently been told to favor `mint` because `forge` reminds people of `forgery` :/ All of these words might have slightly different connotations (eg I wonder if `issue` is specifically related to public keys) in the blockchain domain that I have yet to come to appreciate.

I at least revised these comments to not juxtapose different terms. I'm glad you mentioned it.